### DISCUSSION TITLE
How to solve the problem of WcaSetProperty or MsiSetPropertyA function about dailog not taking effect if not closed

### DISCUSSION THREAD

**Discussion by ad-chuliang** [wixtoolset/issues#8652] [Questions]
I use CustomAction to change the bind value of Property. In C++ function, like "hr = WcaSetProperty(L"FreeSize", UlonglongToLPWSTR(freeSize.QuadPart / (1024 * 1024)));" to set Property.  This statement run indeedly when I bind Publish to control, however, the property value is not change when the Dialog where this control is located is not yet closed. When I close this Dialog,without run that statement again, the property value is changed. So, what's the reason for this situation and how to refresh property value without close its dialog.

**Comment by bevanweiss** [comment]
Are you sure it's the Property value that is not changing, or just the value displayed on the Dialog from the Property?

My recollection was that the Windows Installer Dialogs only update values immediately prior to display.  Once a particular dialog is displayed, the values shown are relatively fixed.
The way I've always gotten it to update is to reshow the same dialog again.

**Reply by ad-chuliang** [reply]
![1](https://github.com/user-attachments/assets/ea8125b2-6425-4604-af75-300e729d9013)
![2](https://github.com/user-attachments/assets/a19c67bb-5f66-47f6-9ead-e8788599fec4)
![3](https://github.com/user-attachments/assets/ddb583d1-5bd8-43e7-8186-9340b3b484e6)
![4](https://github.com/user-attachments/assets/61fb70cb-0a11-450e-a0c2-465ed9890759)
Only when I close the "Disk Space" dialog, the property value will change.
Do you mean I have to re-open the "Disk Space" dialog? 
For example: 
<Dialog Id = "DiskDlg" ... >
  <Control Id="Ok" Type="PushButton" ... >
    <Publish Event="EndDialog" Value="Return"/>
    <Publish Event="NewDialog" Value="DiskDlg"/>
  </Control>
</Dialog>
